src/lib/firestore-schema.ts

```diff
@@ -0,0 +1,152 @@
+type FSDate = Date;
+
+/*
+ * users
+ *
+ * Subcollection of domain with some fields copied from portal (e.g. name) with potential synchronization issues.
+ * At least initially, the list of networks a teacher is a part of will be maintained manually by editing in the
+ * Firestore console. It is anticipated that the numbers of teachers and networks will be small initially.
+ */
+interface UserDocument {
+  uid: string;                // portal user id
+  name: string;               // teacher name
+  type: string;               // "teacher" for now although maybe students eventually
+  networks: string[];         // ids of networks this teacher is part of
+}
+// collection key is portal uid
+type UsersCollection = FSCollection<UserDocument>;
+
+/*
+ * comments
+ *
+ * Subcollection of documents with a denormalized teacher name field with potential synchronization issues.
+ * Comments associated with a document are accessible to any user with access to the document.
+ */
+interface CommentDocument {
+  uid: string;                // user id of author of comment
+  name: string;               // [denormalized] name of user to avoid having to request user data separately
+  createdAt: FSDate;          // timestamp used for sorting
+  tileId?: string;            // empty for document contents
+  content: string;            // plain text for now; potentially html if we need rich text
+}
+// collection key is Firestore-assigned id
+type CommentsCollection = FSCollection<CommentDocument>;
+
+/*
+ * documents
+ *
+ * Subcollection of domain with several metadata fields copied from real-time database (e.g. title, properties)
+ * with potential synchronization issues.
+ * Documents and their comments are accessible to the owning teacher in addition to any network access.
+ * Documents owned by one teacher are accessible to all teachers that share a network with the owner.
+ * Networked access will be mediated by a Firebase function which can check whether the requesting user and
+ * the owning user share a network.
+ */
+interface DocumentDocument {
+  uid: string;
+  type: string;
+  key: string;
+  createdAt: FSDate;
+  title?: string;
+  properties: Record<string, string>;
+  comments: CommentsCollection;
+}
+// collection key is id in Firebase Real-time Database
+type DocumentsCollection = FSCollection<DocumentDocument>;
+
+/*
+ * classes
+ *
+ * Subcollection of domain with several fields copied from portal (e.g. name, teachers) with
+ * potential synchronization issues.
+ * Class resources are accessible to other teachers that share a network with one of the teachers of the class.
+ * Networked access will be mediated by a Firebase function which can check whether the requesting user and
+ * one of the teachers of the class share a network.
+ */
+interface ClassDocument {
+  id: string;                 // portal class id
+  name: string;               // portal class name
+  hash: string;               // portal class hash
```

Comment:
I'd go with context_id here it is more a LTI'ish name and what we are trying to move towards. It's value would be the portal class hash (until that gets renamed to context_id).